bigtable/google/cloud/bigtable/batcher.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""User friendly container for Google Cloud Bigtable MutationBatcher."""
+
+
+import sys
+
+FLUSH_COUNT = 1000
+MAX_MUTATIONS = 100000
+MAX_ROW_BYTES = 5242880  # 5MB
+
+
+class MutationsBatcher(object):
+    """ Batch mutations using limits MAX_ROW_BYTES, MAX_MUTATIONS and
```

Comment:
Please change this comment to something like:

" A MutationsBatcher is used in batch cases where the number of mutations is large or unknown.  It will store `DirectRow`s in memory until one of the size limits is reached, or an explicit call to `flush()` is performed.  When a flush event occurs, the `DirectRow`s in memory will be sent to Cloud Bigtable.  Batching mutations is more efficient than sending individual req

This class is not suited for usage in systems where each mutation needs to guaranteed to be sent, since calling `mutate` may only result in an in-memory change.  In a case of a system crash, any `DirectRow`s remaining in memory will not necessarily be sent to the service, even after the completion of the `mutate()` method.

TODO: Performance would dramatically improve if this class had the capability of asynchronous, parallel RPCs. 
"